checker/env.go

```diff
@@ -212,14 +217,14 @@ func (e *Env) addFunction(decl *exprpb.Decl) []errorMsg {
 	if current == nil {
 		//Add the function declaration without overloads and check the overloads below.
 		current = decls.NewFunction(decl.Name)
-		e.declarations.AddFunction(current)
+	} else {
+		// Copy on write since we don't know where this original definition came from.
+		current = proto.Clone(current).(*exprpb.Decl)
 	}
+	e.declarations.SetFunction(current)
 
 	errorMsgs := make([]errorMsg, 0)
 	for _, overload := range decl.GetFunction().GetOverloads() {
-		if _, found := e.filteredOverloadIDs[overload.GetOverloadId()]; found {
-			continue
-		}
```

Comment:
Just to double check. Removing this code is intentional because it allows for set semantics here, right?

Reply:
I simply moved the filtering to the `checker.resolveOverloadOrErr` calls, ensuring that disabled (filtered) overloads are not in the set considered during type-checking.